### ISSUE TITLE
error: redefinition of local "$my_local"

### ISSUE THREAD

**Issue by paulyoung** [WebAssembly/wabt#966]
If I try to convert from wasm to wat and back to wasm again, like this:

```
$ wasm2wat foo.wasm > foo.wat
$ wat2wasm foo.wat > foo2.wasm
```

I get errors saying:  `error: redefinition of local "$my_local"`

@nomeata pointed out to me that:

> The names for locals are not unique (and need not be, [per spec](https://webassembly.github.io/spec/core/appendix/custom.html), “Names need not be unique.”).

@rossberg suggested that I try the spec interpreter instead, and create an issue here for duplicate symbols being flagged as an error.

```
$ wasm -d foo.wasm -o foo2.wasm
```

**Comment by binji** [comment]
Heh, looks like we fixed this a while back for functions (see PR #512), but not for locals. :-)

I added a PR to fix this in #968.

**Comment by paulyoung** [comment]
Thanks @binji!
